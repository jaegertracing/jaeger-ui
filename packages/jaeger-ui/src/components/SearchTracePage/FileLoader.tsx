// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { Upload } from 'antd';
import { IoDocumentAttachOutline } from 'react-icons/io5';

import './FileLoader.css';

const Dragger = Upload.Dragger;

type FileLoaderProps = {
  loadJsonTraces: (fileList: { file: File }) => void;
};

export default function FileLoader(props: FileLoaderProps) {
  return (
    <Dragger
      accept=".json"
      beforeUpload={(file, fileList) =>
        fileList.forEach(fileFromList => props.loadJsonTraces({ file: fileFromList }))
      }
      multiple
    >
      <IoDocumentAttachOutline className="Dragger--icon" />
      <p className="ant-upload-text">Click or drag files to this area.</p>
      <p className="ant-upload-hint">JSON files containing one or more traces are supported.</p>
    </Dragger>
  );
}
