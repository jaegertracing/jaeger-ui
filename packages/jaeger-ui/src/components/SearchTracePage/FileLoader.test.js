// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import { shallow } from 'enzyme';

import FileLoader from './FileLoader';

describe('<FileLoader />', () => {
  let wrapper;
  const mockLoadJsonTraces = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<FileLoader loadJsonTraces={mockLoadJsonTraces} />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls loadJsonTraces with the uploaded file', () => {
    const file = new File(['sample content'], 'sample.json', { type: 'application/json' });
    const fileList = [file];

    wrapper.find('Dragger').prop('beforeUpload')(file, fileList);

    expect(mockLoadJsonTraces).toHaveBeenCalledWith({ file });
  });
});
