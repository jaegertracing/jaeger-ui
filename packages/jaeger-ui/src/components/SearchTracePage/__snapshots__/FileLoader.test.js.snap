// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileLoader /> matches the snapshot 1`] = `
<Dragger
  accept=".json"
  multiple={true}
>
  <p
    className="ant-upload-drag-icon"
  >
    <Icon
      type="file-add"
    />
  </p>
  <p
    className="ant-upload-text"
  >
    Click or drag files to this area.
  </p>
  <p
    className="ant-upload-hint"
  >
    Support JSON files containig one or more traces.
  </p>
</Dragger>
`;
